From ad0fcef3b70c3f07739737dc60d6e60ea1523726 Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Tue, 1 Jul 2025 12:19:42 +0100
Subject: [PATCH] plugins/cilium-cni/install-plugin.sh: install portable
 OpenSSL

In plugins/cilium-cni/install-plugin.sh install portable OpenSSL along
side the CNI binaries.
---
 plugins/cilium-cni/install-plugin.sh | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/plugins/cilium-cni/install-plugin.sh b/plugins/cilium-cni/install-plugin.sh
index dc99174de6..1b35bbb684 100755
--- a/plugins/cilium-cni/install-plugin.sh
+++ b/plugins/cilium-cni/install-plugin.sh
@@ -33,3 +33,17 @@ if [ ! -f "${CNI_DIR}/bin/loopback" ]; then
 fi
 
 install_cni "/opt/cni/bin/${BIN_NAME}"
+
+# Always install cgr portable OpenSSL
+
+CGR_DIR=${CNI_DIR}/bin/.cgr
+mkdir -p ${CGR_DIR}
+for src in /etc/ssl/fipsmodule.cnf /etc/ssl/openssl.cnf /usr/lib/libcrypto3/libcrypto.so.3 /usr/lib/ossl-modules/fips.so; do
+	bin_name="$(basename $src)"
+	tmp_dst="${CGR_DIR}/.$bin_name.new"
+	dst="${CGR_DIR}/$bin_name"
+	echo "Installing $bin_name to $dst ..."
+	cp $src $tmp_dst && \
+	mv $tmp_dst $dst && \
+	echo "Wrote $dst"
+done
-- 
2.48.1

